examples/ocean-dbtc-burning-bot/src/main.ts

```diff
@@ -0,0 +1,160 @@
+const PRIVATE_KEY = 'PRIVATE_KEY_IN_WIF'
+
+export async function main (): Promise<void> {
+  const wallet = new WalletClassic(WIF.asEllipticPair(PRIVATE_KEY))
+  const sequencer = new BurnProgram(client, wallet)
+
+  // Refill UTXO if balance is getting low
+  if (await sequencer.accountToUTXO()) {
+    return
+  }
+
+  // Burn dBTC by sending to burn address
+  if (await sequencer.burnDBTC()) {
+    return
+  }
+
+  // Swap DFI to BTC
+  await sequencer.swapDFIDBTC()
+}
+
+export class BurnProgram {
+  constructor (
+    private readonly client: WhaleApiClient,
+    private readonly wallet: WalletClassic,
+    private readonly account = new WhaleWalletAccount(client, wallet, MainNet)
+  ) {
+  }
+
+  /**
+   * Refill UTXO when there is less than 1.1 UTXO.
+   */
+  async accountToUTXO (): Promise<boolean> {
+    const utxoBalance = await this.getUTXOBalance()
+    console.log(`UTXO Balance: ${utxoBalance.toFixed()}`)
+
+    if (utxoBalance.gte(new BigNumber('1.1'))) {
+      return false
+    }
+    console.log('Refilling UTXO')
+
+    const script = await this.account.getScript()
+    const txn = await this.account.withTransactionBuilder().account.accountToUtxos({
+      from: script,
+      balances: [
+        {
+          token: 0,
+          amount: new BigNumber('1.0')
+        }
+      ],
+      mintingOutputsStart: 2 // 0: DfTx, 1: change, 2: minted utxos (mandated by jellyfish SDK)
+    }, script)
+
+    await this.send(txn)
+    return true
+  }
+
+  /**
+   * Burn dBTC there is more than 0.1 BTC
+   */
+  async burnDBTC (): Promise<boolean> {
+    const dBTC = await this.getTokenBalance('BTC', new BigNumber('0.1'))
+    if (dBTC === undefined) {
+      return false
+    }
+    console.log(`Burning dBTC: ${dBTC.amount}`)
+
+    const script = await this.account.getScript()
+    const txn = await this.account.withTransactionBuilder().account.accountToAccount({
+      from: script,
+      to: [{
+        script: BURN_ADDRESS_SCRIPT,
+        balances: [{
+          token: parseInt(dBTC.id),
+          amount: new BigNumber(dBTC.amount)
+        }]
+      }]
+    }, script)
+
+    await this.send(txn)
+    return true
+  }
+
+  /**
+   * Swap DFI to dBTC when there is more than 100 DFI
+   */
+  async swapDFIDBTC (): Promise<boolean> {
+    const dfi = await this.getTokenBalance('DFI', new BigNumber('100'))
+    if (dfi === undefined) {
+      console.log('DFI Account balance less than 100')
+      return false
+    }
+    console.log(`Swap ${dfi.amount} DFI to dBTC`)
+
+    const script = await this.account.getScript()
+    const txn = await this.account.withTransactionBuilder().dex.compositeSwap({
+      poolSwap: {
+        fromScript: script,
+        toScript: script,
+        fromTokenId: 0,
+        toTokenId: 2,
+        fromAmount: new BigNumber(dfi.amount),
+        maxPrice: new BigNumber('9223372036854775807')
+      },
+      pools: []
```

Comment:
I think `poolId` for sending raw `CompositeSwap` is mandatory

Reply:
https://github.com/DeFiCh/ain/pull/956/files#diff-5b5f5b126aad5172f5e15066f2926f4afb57e75eb624b509f56233c14cd4b5c9R1178-R1182

No hops, don't need to specify, only 2-3 hops required poolId.